test/test_infer_pieces.py

```diff
@@ -0,0 +1,236 @@
+"""This module is responsible for testing "infer_pieces.py" module.
+
+Specifically, it tests the `_determine_changed_squares()` and
+`_detect_move()` functions in the module.
+"""
+
+
+from lc2fen.fen import fen_to_board, board_to_list
+from lc2fen.infer_pieces import (
+    _PIECE_TO_IDX_FULL,
+    _determine_changed_squares,
+    _detect_move,
+)
+
+
+def generate_probs_with_no_indices_from_fen(fen: str):
+    """Generate `probs_with_no_indices` from FEN string.
+
+    This function takes a FEN string as input and generates the
+    hardcoded `probs_with_no_indices`.
+
+    :param fen: FEN string of the board position of interest.
+
+    :return: Length-64 list of piece probabilities.
+
+        Each element in the list is a length-13 sublist that corresponds
+        to a unique square on the chessboard.
+
+        Each sublist contains 13 piece probabilities (in the order of
+        `_IDX_TO_PIECE_FULL`) for the corresponding square.
+
+        The probabilities are hardcoded such that they are all 0s and
```

Comment:
Could you add in the doc that the one will be exactly the piece set by the fen string?

Reply:
Done! Lmk what you think

Reply:
I would simplify it a bit more. How about:
The probabilities are hardcoded such that each sublist has exactly twelve 0s and one 1. The 1 will correspond to the piece type given by the fen string.

Reply:
How does this sound?
![image](https://github.com/davidmallasen/LiveChess2FEN/assets/51491203/96fceb3e-c440-4f6b-aa13-64c86ab52a1d)